internal/configs/transportserver.go

```diff
@@ -29,10 +49,10 @@ func (tsEx *TransportServerEx) String() string {
 }
 
 // generateTransportServerConfig generates a full configuration for a TransportServer.
-func generateTransportServerConfig(transportServerEx *TransportServerEx, listenerPort int, isPlus bool) version2.TransportServerConfig {
+func generateTransportServerConfig(transportServerEx *TransportServerEx, listenerPort int, tsc *transportServerConfigurator) version2.TransportServerConfig {
```

Comment:
this really looks a method of `transportServerConfigurator`. Could we convert the function to a method rather than passing `*transportServerConfigurator` as an argument?

Reply:
I think it's best to keep these types of functions as is. 

A method operates on a struct, which can hold state. If the function is stateless, then we can more easily reason about and test the function. When we provide specific inputs, we always get the same outputs. If it converted to a method, then the outputs also depend on the state of the struct, which can change and is shared.

Reply:
in that case, I think it's better to call it something like `transportServerConfigurationOptions`, not `transportServerConfigurator`. 

Also, warnings in that case -- better to make `generateTransportServerConfig` return them - rather than store them in `transportServerConfigurator`. Otherwise, we'll have a type that acts both as input and output, which also makes it  harder to reason about it.

Reply:
done